service/src/main/java/org/apache/griffin/core/job/JobServiceImpl.java

```diff
@@ -158,17 +162,22 @@ public JobServiceImpl() {
         } catch (SchedulerException e) {
             LOGGER.error("Failed to get RUNNING jobs.", e);
             throw new GriffinException
-                    .ServiceException("Failed to get RUNNING jobs.", e);
+                .ServiceException("Failed to get RUNNING jobs.", e);
         }
         return dataList;
     }
 
     @Override
     public AbstractJob addJob(AbstractJob job) throws Exception {
+        JobEvent jobEvent = JobEvent.yieldJobEventBeforeCreation(null);
```

Comment:
shouldn't job be passed here? this would allow to validate some job parameters, for example

Reply:
sure, let me change it

Reply:
@Test(expected=UnImplementedException.class), yeah I ever have considered. but hope not to throw exception in a transaction, so make it empty method.

Reply:
@toyboxman Can we verify it by assert verify on called? instead of some exception throwed.
currently test case , not straightforward to me.